tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/SecondaryIndexTable.java

```diff
@@ -177,11 +173,23 @@ public void close() throws IOException {
     } catch (IOException e) {
       try {
         secondaryIndexTable.close();
+        conn.close();
       } catch (IOException ex) {
         e.addSuppressed(e);
       }
       throw e;
     }
     secondaryIndexTable.close();
+    conn.close();
+  }
+
+  protected void finalize() throws Throwable {
```

Comment:
We try not to rely on `finalize()` since there is no guarantee when it will be called. Also since we are closing the connection in the `close()` method, there is no need for finalize. Let's remove this method.